owncloudData/src/test/java/com/owncloud/android/data/webfinger/datasource/implementation/OCRemoteWebFingerDatasourceTest.kt

```diff
@@ -0,0 +1,150 @@
+                resource = OC_SECURE_SERVER_INFO_BASIC_AUTH.baseUrl,
+                rel = WebFingerRel.OIDC_ISSUER_DISCOVERY.uri,
+                ownCloudClient,
+            )
+        } returns getInstancesFromWebFingerResult
+
+        val actualResult = ocRemoteWebFingerDatasource.getInstancesFromWebFinger(
+            lookupServer = OC_SECURE_SERVER_INFO_BASIC_AUTH.baseUrl,
+            rel = WebFingerRel.OIDC_ISSUER_DISCOVERY,
+            resource = OC_SECURE_SERVER_INFO_BASIC_AUTH.baseUrl,
+        )
+
+        assertEquals(getInstancesFromWebFingerResult.data, actualResult)
+
+        verify(exactly = 1) {
+            clientManager.getClientForAnonymousCredentials(any(), false)
+            ocWebFingerService.getInstancesFromWebFinger(
+                lookupServer = OC_SECURE_SERVER_INFO_BASIC_AUTH.baseUrl,
+                resource = OC_SECURE_SERVER_INFO_BASIC_AUTH.baseUrl,
+                rel = WebFingerRel.OIDC_ISSUER_DISCOVERY.uri,
+                ownCloudClient,
+            )
+        }
+    }
+
+    @Test(expected = Exception::class)
+    fun `getInstancesFromWebFinger returns an exception when service receive an exception`() {
+
+        every {
+            ocWebFingerService.getInstancesFromWebFinger(
+                lookupServer = OC_SECURE_SERVER_INFO_BASIC_AUTH.baseUrl,
+                resource = OC_SECURE_SERVER_INFO_BASIC_AUTH.baseUrl,
+                rel = WebFingerRel.OIDC_ISSUER_DISCOVERY.uri,
+                any(),
+            )
+        } throws Exception()
+
+        ocRemoteWebFingerDatasource.getInstancesFromWebFinger(
+            lookupServer = OC_SECURE_SERVER_INFO_BASIC_AUTH.baseUrl,
+            rel = WebFingerRel.OIDC_ISSUER_DISCOVERY,
+            resource = OC_SECURE_SERVER_INFO_BASIC_AUTH.baseUrl,
+        )
+    }
+
+    @Test
+    fun `getInstancesFromAuthenticatedWebFinger returns a list of web finger`() {
+
+        val getInstancesFromAuthenticatedWebFingerResult: RemoteOperationResult<List<String>> =
+            createRemoteOperationResultMock(data = listString, isSuccess = true)
+
+        every {
+            ocWebFingerService.getInstancesFromWebFinger(
+                lookupServer = OC_SECURE_SERVER_INFO_BASIC_AUTH.baseUrl,
+                resource = OC_SECURE_SERVER_INFO_BASIC_AUTH.baseUrl,
+                rel = WebFingerRel.OIDC_ISSUER_DISCOVERY.uri,
+                ownCloudClient,
+            )
+        } returns getInstancesFromAuthenticatedWebFingerResult
+
+        val actualResult = ocRemoteWebFingerDatasource.getInstancesFromAuthenticatedWebFinger(
+            lookupServer = OC_SECURE_SERVER_INFO_BASIC_AUTH.baseUrl,
+            rel = WebFingerRel.OIDC_ISSUER_DISCOVERY,
+            resource = OC_SECURE_SERVER_INFO_BASIC_AUTH.baseUrl,
+            username = OC_ACCOUNT_ID,
+            accessToken = OC_ACCESS_TOKEN
+        )
+
+        assertEquals(getInstancesFromAuthenticatedWebFingerResult.data, actualResult)
+
+        verify(exactly = 1) {
```

Comment:
Try to verify in this one that the `credentials` property is set somehow, because it's the only thing that differentiates from the other test. Otherwise, we would have 2 equals tests 👎

Reply:
Done